api/v1/installation_types.go

```diff
@@ -129,17 +129,30 @@ type InstallationSpec struct {
 type TyphaAffinity struct {
 	// NodeAffinity describes node affinity scheduling rules for typha.
 	// +optional
-	NodeAffinity *PreferredNodeAffinity `json:"nodeAffinity,omitempty"`
+	NodeAffinity *TigeraNodeAffinty `json:"nodeAffinity,omitempty"`
 }
 
-// PreferredNodeAffinity is similar to an affinity except it only exposes the Preferred scheduling option,
-// which ensures a pod will still be scheduled even when all the nodes it matches are unschedulable.
-type PreferredNodeAffinity struct {
+// TigeraNodeAffinty is similar to *v1.NodeAffinity, but allows us to limit available shcedulers.
+type TigeraNodeAffinty struct {
 	// The scheduler will prefer to schedule pods to nodes that satisfy
 	// the affinity expressions specified by this field, but it may choose
 	// a node that violates one or more of the expressions.
 	// +optional
 	PreferredDuringSchedulingIgnoredDuringExecution []v1.PreferredSchedulingTerm `json:"preferredDuringSchedulingIgnoredDuringExecution,omitempty"`
+
+	// WARNING: Please note that if the affinity requirements specified by this field are not met at
+	// scheduling time, the pod will NOT be scheduled onto the node.
+	// There is no fallback to another affinity rules with this setting.
+	// This may cause networking disruption or even catastrophic failure!
+	// PreferredDuringSchedulingIgnoredDuringExecution should be a default choice if
```

Comment:
I do not think this sentence would be very clear to someone that was not familiar with what we are trying to convey. How about 
```
PreferredDuringSchedulingIgnoredDuringExecution should be used for affinity unless there is a specific well understood reason to use RequiredDuringSchedulingIgnoredDuringExecution and you can guarantee that the RequiredDuringSchedulingIgnoredDuringExecution will always have sufficient nodes to satisfy the requirement.
```